src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/PreferStreamReadAsyncMemoryOverloadsTests.cs

```diff
@@ -400,234 +443,223 @@ End Class
+                    false, 70)]
+        [InlineData("New Byte(s.Length - 1) {}, 0, s.Length",
+                    "(New Byte(s.Length - 1) {}).AsMemory(0, s.Length)",
+                    true, 70)]
+        [InlineData("New Byte(s.Length - 1) {}, 0, s.Length, New CancellationToken()",
+                    "(New Byte(s.Length - 1) {}).AsMemory(0, s.Length), New CancellationToken()",
+                    false, 95)]
+        [InlineData("New Byte(s.Length - 1) {}, 0, s.Length, New CancellationToken()",
+                    "(New Byte(s.Length - 1) {}).AsMemory(0, s.Length), New CancellationToken()",
+                    true, 95)]
+        public Task VB_Analyzer_Diagnostic_InlineByteArray(string originalArgs, string fixedArgs, bool withConfigureAwait, int endColumn)
+        {
+            string source = @"
 Imports System
 Imports System.IO
 Imports System.Threading
-Class C
+Public Module C
     Public Async Sub M()
-        Using fs As FileStream = File.Open(""file.txt"", FileMode.Open)
-            Await fs.ReadAsync(New Byte(fs.Length - 1) {}, 0, fs.Length)
+        Using s As FileStream = File.Open(""file.txt"", FileMode.Open)
+            Await {0}
         End Using
     End Sub
-End Class
-            ", GetVBResult(8, 19, 8, 73));
+End Module
+            ";
+            return GetVBReadAsyncDiagnostic(source, originalArgs, fixedArgs, withConfigureAwait, 8, 19, 8, endColumn);
         }
 
-        [Fact]
-        public Task VB_Analyzer_Diagnostic_CancellationToken()
-        {
-            return AnalyzeVBAsync(@"
-Imports System
-Imports System.IO
-Imports System.Threading
-Class C
-    Public Async Sub M()
-        Using fs As FileStream = File.Open(""file.txt"", FileMode.Open)
-            Dim buffer As Byte() = New Byte(fs.Length - 1) {}
-            Await fs.ReadAsync(buffer, 0, fs.Length, New CancellationToken())
-        End Using
-    End Sub
-End Class
-            ", GetVBResult(9, 19, 9, 78));
-        }
+        #endregion
 
-        [Fact]
-        public Task VB_Analyzer_Diagnostic_ConfigureAwait()
+        #region Helpers
+
+        private const string AsyncMethodName = "Read";
+
+        private Task GetCSReadAsyncDiagnostic(string source, string originalArgs, string fixedArgs, bool withConfigureAwait, int startLine, int startColumn, int endLine, int endColumn)
         {
-            return AnalyzeVBAsync(@"
-Imports System
-Imports System.IO
-Imports System.Threading
-Class C
-    Public Async Sub M()
-        Using fs As FileStream = File.Open(""file.txt"", FileMode.Open)
-            Dim buffer As Byte() = New Byte(fs.Length - 1) {}
-            Await fs.ReadAsync(buffer, 0, fs.Length).ConfigureAwait(False)
-        End Using
-    End Sub
-End Class
-            ", GetVBResult(9, 19, 9, 75));
+            return FixCSAsync(
+                GetFormattedSourceCode(source, AsyncMethodName, originalArgs, withConfigureAwait, LanguageNames.CSharp),
```

Comment:
None of our unit tests need to do any formatting of input/output source code. Why is this needed?

Reply:
Ah, so I read the implementation of `GetFormattedSourceCode` and the helper is not actually performing whitespace formatting, but producing input/output code formatted based on the arguments passed in here. Seems reasonable, but the name might confuse some readers.

Reply:
Let me change the name for clarity